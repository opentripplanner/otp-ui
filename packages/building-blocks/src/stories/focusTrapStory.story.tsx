import React from "react";
import FocusTrapWrapper from "../focus-trap-wrapper";

export default {
  component: FocusTrapWrapper,
  title: "Building-Blocks/FocusTrapWrapper"
};

export const FocusTrapAroundButtonSet = (): React.ReactElement => (
  <FocusTrapWrapper id="button-set-story">
    <button type="button">Button 1</button>
    <button type="button">Button 2</button>
    <button type="button">Button 3</button>
    <button type="button">Button 4</button>
  </FocusTrapWrapper>
);

export const FocusTrapWithVariousEls = (): React.ReactElement => (
  <FocusTrapWrapper
    focusElements={["button", "a", "div", "input", "select"]}
    id="various-els-story"
  >
    <button type="button">Button 1</button>
    <br />
    <a href="/">link</a>
    <br />
    <div tabIndex={-1}>focusable div</div>
    <label htmlFor="disabled-input">
      Input (disabled) <input id="disabled-input" disabled type="text" />{" "}
    </label>

    <br />
    <label htmlFor="input">
      Input <input id="input" type="text" />
    </label>

    <br />
    <label htmlFor="select">
      Select{" "}
      <select id="select">
        <option>Option 1</option>
        <option>Option 2</option>
      </select>
    </label>
  </FocusTrapWrapper>
);
