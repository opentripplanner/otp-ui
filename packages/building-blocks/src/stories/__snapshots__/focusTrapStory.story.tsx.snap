// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building-Blocks/FocusTrapWrapper FocusTrapAroundButtonSet smoke-test 1`] = `
<div id="button-set-story-focus-trap"
     role="presentation"
>
  <button type="button">
    Button 1
  </button>
  <button type="button">
    Button 2
  </button>
  <button type="button">
    Button 3
  </button>
  <button type="button">
    Button 4
  </button>
</div>
`;

exports[`Building-Blocks/FocusTrapWrapper FocusTrapWithVariousEls smoke-test 1`] = `
<div id="various-els-story-focus-trap"
     role="presentation"
>
  <button type="button">
    Button 1
  </button>
  <br>
  <a href="/">
    link
  </a>
  <br>
  <div tabindex="-1">
    focusable div
  </div>
  <label for="disabled-input">
    Input (disabled)
    <input id="disabled-input"
           disabled
           type="text"
    >
  </label>
  <br>
  <label for="input">
    Input
    <input id="input"
           type="text"
    >
  </label>
  <br>
  <label for="select">
    Select
    <select id="select">
      <option>
        Option 1
      </option>
      <option>
        Option 2
      </option>
    </select>
  </label>
</div>
`;
